hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -1626,6 +1638,22 @@ func (r *NodePoolReconciler) enqueueNodePoolsForConfig(obj client.Object) []reco
 		return enqueueParentNodePool(obj)
 	}
 
+	// Check if the ConfigMap is generated by an operator in the control plane namespace
+	// corresponding to this nodepool.
+	if _, ok := obj.GetLabels()[nodePoolOperatorCreatedConfigLabel]; ok {
+		// Go through the list of NodePools to determine which one
+		// this ConfigMap is referencing
+		for key := range nodePoolList.Items {
+			if obj.GetAnnotations()[nodePoolAnnotation] == nodePoolList.Items[key].Name &&
+				nodePoolList.Items[key].GetNamespace() == obj.GetNamespace() {
+				result = append(result, reconcile.Request{
+					NamespacedName: client.ObjectKeyFromObject(&nodePoolList.Items[key]),
+				})
+				break
+			}
+		}
```

Comment:
@enxebre ideally here we could enqueueParentNodePool for these ConfigMaps, but they are labeled with the NodePool name only, from the NodePool labels on the Nodes. I might be missing a more elegant solution?